From 17ff615d71162a475547cbdc0c052fad57e32cc1 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:06:26 -0400
Subject: [acpi] dock: Make 'docked' sysfs attribute work as documented

Message-id: <20140828030626.2117.25474.stgit@gir.stowe>
Patchwork-id: 89316
O-Subject: [RHEL7.1 PATCH 28/30] ACPI / dock: Make 'docked' sysfs attribute work as documented
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: ab62f9cd7b5ecdf853f1612fe1e983cb7cbbac3e

commit ab62f9cd7b5ecdf853f1612fe1e983cb7cbbac3e
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sat Feb 15 01:29:06 2014 +0100

    ACPI / dock: Make 'docked' sysfs attribute work as documented

    After recent ACPI core changes acpi_bus_get_device() will always
    succeed for dock station ACPI device objects, so show_docked()
    should not use that function's return value as an indicator of
    whether or not the dock device is present.

    Make it use acpi_device_enumerated() for this purpose.

    Fixes: 202317a573b2 (ACPI / scan: Add acpi_device objects for all device nodes in the namespace)
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index 4cbc0bd..5fb301f 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -715,13 +715,11 @@ static acpi_status __init find_dock_devices(acpi_handle handle, u32 lvl,
 static ssize_t show_docked(struct device *dev,
       struct device_attribute *attr, char *buf)
 {
- struct acpi_device *tmp;
-
  struct dock_station *dock_station = dev->platform_data;
+ struct acpi_device *adev = NULL;
 
- if (!acpi_bus_get_device(dock_station->handle, &tmp))
-  return snprintf(buf, PAGE_SIZE, "1\n");
- return snprintf(buf, PAGE_SIZE, "0\n");
+ acpi_bus_get_device(dock_station->handle, &adev);
+ return snprintf(buf, PAGE_SIZE, "%u\n", acpi_device_enumerated(adev));
 }
 static DEVICE_ATTR(docked, S_IRUGO, show_docked, NULL);
 
-- 
1.7.1